### PR TITLE
Adding Image support

### PR DESCRIPTION
**Pull request by VasudhaMathur**

### CODE REVIEW

File: src/ruqola.h [hunk 1]
```diff
@@ -67,20 +68,22 @@ class Ruqola: public QObject
...
+
 //     QString activeRoom() const;
 //     void setActiveRoom(const QString &activeRoom);
 
     DDPClient *ddp();
     Notification * notification();
 
     Q_INVOKABLE RoomModel *roomModel();
-    Q_INVOKABLE void sendMessage(const QString &roomID, const QString &message);
+    Q_INVOKABLE void sendMessage(const QString &roomID, const QString &message, const QString type);
```
Comment (ruphy) [inline src/ruqola.h]: const QString &type

File: src/ruqola.h [hunk 2]
```diff
@@ -26,6 +26,7 @@
 #include <QObject>
 #include <QQmlEngine>
 #include <QJSEngine>
+#include <QTcpSocket>
```
Comment (ruphy) [inline src/ruqola.h]: Do you need this here?

File: src/ruqola.cpp [hunk 3]
```diff
@@ -106,12 +109,39 @@ Notification * Ruqola::notification()
...
+}
 
-void Ruqola::sendMessage(const QString &roomID, const QString &message)
+void Ruqola::sendMessage(const QString &roomID, const QString &message, const QString type)
 {
-    QString json = "{\"rid\": \"%1\", \"msg\": \"%2\"}";
-    json = json.arg(roomID, message);
-    ddp()->method("sendMessage", QJsonDocument::fromJson(json.toUtf8()));
+//    QString json = "{\"rid\": \"%1\", \"msg\": \"%2\", \"type\": \"%3\"}";
+//    json = json.arg(roomID, message, type);
```
Comment (ruphy) [inline src/ruqola.cpp]: great!

File: src/ruqola.cpp [hunk 4]
```diff
@@ -106,12 +109,39 @@ Notification * Ruqola::notification()
...
+//    qDebug() << "Sending json: " << json;
+    QJsonObject jSonImage;
+    jSonImage.insert("rid",roomID);
+    jSonImage.insert("msg",message);
+    jSonImage.insert("type",type);
```
Comment (ruphy) [inline src/ruqola.cpp]: you can also use, in short:
`jsonImage["rid"] = roomID;`

Also, it's json, not jSon

File: src/ruqola.cpp [hunk 5]
```diff
@@ -106,12 +109,39 @@ Notification * Ruqola::notification()
     return m_notification;
 }
 
+void Ruqola::attachmentButtonClicked()
```
Comment (ruphy) [inline src/ruqola.cpp]: it's OK for a test but we will re-write this function later

File: src/ddpclient.cpp [hunk 6]
```diff
@@ -170,22 +171,67 @@ void DDPClient::onTextMessageReceived(QString message)
...
+                //check message type
+                QJsonDocument res = QJsonDocument(root.value("result").toObject());
+                QJsonObject result = res.object();
+                QString type = result.value("type").toString();
+                QString msg = result.value("msg").toString();
+
+                QByteArray base64Image;
+                QImage image;
+
+                QString path = "/home/vasudha/Qt/build-Ruqola-Desktop_Qt_5_8_0_GCC_64bit-Debug/Images";
```
Comment (ruphy) [inline src/ddpclient.cpp]: please change this directory with somehting from QStandardPaths (e.g. cache)

File: Ruqola.pro.user [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE QtCreatorProject>
-<!-- Written by QtCreator 4.2.1, 2017-06-01T11:14:33. -->
+<!-- Written by QtCreator 4.2.1, 2017-06-02T22:56:25. -->
```
Comment (ruphy) [inline Ruqola.pro.user]: don't commit this file: remember it or add it to .gitignore if you want

File: src/messagemodel.cpp [hunk 8]
```diff
@@ -51,14 +68,31 @@ QByteArray MessageModel::serialize(const Message& message)
...
+    o["userID"] = message.userID;
+    o["updatedAt"] = message.updatedAt;
+    o["editedAt"] = message.editedAt;
+    o["editedByUsername"] = message.editedByUsername;
+    o["editedByUserID"] = message.editedByUserID;
+    o["url"] = message.url;
+    o["meta"] = message.meta;
+    o["headers"] = message.headers;
+    o["parsedUrl"] = message.parsedUrl;
+    o["image_url"] = message.image_url;
```
Comment (ruphy) [inline src/messagemodel.cpp]: use CamelCase, for consistency with the rest of the codebase

File: src/messagemodel.h [hunk 9]
```diff
@@ -65,7 +109,22 @@ class MessageModel : public QAbstractListModel
         Timestamp,
         UserID,
         SystemMessage,
-        SystemMessageType
+        SystemMessageType,
+        MessageID,
+        RoomID,
+        UpdatedAt,
+        editedAt,
```
Comment (ruphy) [inline src/messagemodel.h]: Same thing, uniform capitalization, for coding style

File: src/rocketchatbackend.cpp [hunk 10]
```diff
@@ -133,18 +133,33 @@ void RocketChatBackend::processIncomingMessages(QJsonArray messages)
...
+        m.avatar = o.value("avatar").toString();
+        m.groupable = o.value("groupable").toBool();
+        m.parseUrls = o.value("parseUrls").toBool();
+
         if (!type.isEmpty()) {
             m.systemMessage = true;
             m.systemMessageType = type;
         } else {
-            m.systemMessage = false;
+           m.systemMessage = false;
```
Comment (ruphy) [inline src/rocketchatbackend.cpp]: wrong indent, check your diffs before committing :)

File: src/messagemodel.cpp (hunk 8 above)
Comment (VasudhaMathur) [inline src/messagemodel.cpp]: image_url: The image url of the attachment
It's given like this in the Rocket.Chat docs.

**Pull request merged**